Userland/Services/PreviewServer/PreviewProviders/Image.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Image.h"
+#include <LibCore/File.h>
+#include <LibCore/MimeData.h>
+#include <LibGfx/ImageDecoder.h>
+#include <LibGfx/Painter.h>
+
+REGISTER_PREVIEW_PROVIDER(Image)
+
+namespace PreviewServer::Providers {
+
+bool Image::can_generate_preview_for(String const& file)
```

Comment:
not sure which actor should decide on preview size… shall it be a requesting party? (sounds sensible; providing a default, e.g. of proposed 32✖️32 is of course okay)

writing that only because that api is proposed here; on the other hand it does not seem to be carved in stone for the future—feel free to ignore the comment then. (just a another yak food for thought, i guess.)

Reply:
As you said above, we should always use the global constant for the 32. In the future this can be configurable, but remember that I'm going with MVP here.